### PR TITLE
doc(Quickstart): Implementing Suggestions - DoAll and Reduction

### PR DESCRIPTION
**Pull request by goerlibe**
#179

### CODE REVIEW

**Comment by goerlibe** [comment]
My solution for the Geometric Decomposition pattern is wrong and still needs fixing!

- [ ] Correct the loop boundary issues!
- [ ] combine both loops? (in this case it is possible. I think there should be cases when it is not possible. If I am wrong and it is always possible then we should do it, too!)
- [ ] Check formatting of the generated documentation

File: docs/Manual_Quickstart/Manual_Example.md [hunk 1]
```diff
@@ -81,11 +81,48 @@ In order to generate parallelization suggestions from the [gathered data](../Pro
...
+* The first should be a `Do-all` with the `Start line 1:19`.
+* The second should be  `Reduction` with the `Start line 1:23`.
+* The third should be a `Geometric Decomposition`.
+
+## Step 3: Interpreting and Implementing the Suggestions
+For detailed information on how to interpret and implement the created suggestions, please refer to the [pattern wiki](../Pattern_Detection/Patterns/Patterns.md).
+
+Looking at the created suggestions there are two ways that the code can be parallelized (the parallelized code can be found inside the example directory as `solution1.cpp` and `solution2.cpp`):
+
+* **Implementing the Do-All and Reduction Patterns**<br/> To implement the `Do-All` and `Reduction` patterns all we have to do is to add the suggested pragma before each loop. Make sure to add the `reduction` clause when implementing Reductions! The DiscoPoP Explorer also suggests variable classification for used variables to ensure correctness and improve performance. These should be added as clauses to the pragma. Note that OpenMP implicitly makes some variables like the loop index a private variable so we can omit the corresponding clauses.
```
Comment (lukasrothenberger) [inline docs/Manual_Quickstart/Manual_Example.md:93]:
```suggestion
* **Implementing the Do-All and Reduction Patterns**<br/> To implement the `Do-All` and `Reduction` patterns all we have to do is to add the suggested pragma before each loop. Make sure to add the `reduction` clause when implementing Reductions! The DiscoPoP Explorer also suggests classifications for used variables to ensure correctness and improve performance. These should be added as clauses to the pragma. Note that OpenMP implicitly makes some variables like the loop index a private variable so we can omit the corresponding clauses.
```

File: docs/Manual_Quickstart/Manual_Example.md [hunk 2]
```diff
@@ -81,11 +81,48 @@ In order to generate parallelization suggestions from the [gathered data](../Pro
...
+  - To implement the `Do-All` suggestion we add the following line before the corresponding loop.
+      
+        #pragma omp parallel for shared(Arr,N)
+
+  - To implement the `Reduction` suggestion we add the folling line before the corresponding loop.
```
Comment (lukasrothenberger) [inline docs/Manual_Quickstart/Manual_Example.md:98]:
```suggestion
  - To implement the `Reduction` suggestion we add the following line before the corresponding loop.
```

File: docs/Manual_Quickstart/Manual_Example.md [hunk 3]
```diff
@@ -81,11 +81,48 @@ In order to generate parallelization suggestions from the [gathered data](../Pro
...
+    
+        # pragma omp parallel for reduction(+:sum) shared(Arr,N)
+
+  - When we implement both patterns it is better to open only one parallel region with `#pragma omp parallel` and use the `pragma omp for` before each loop:
```
Comment (lukasrothenberger) [inline docs/Manual_Quickstart/Manual_Example.md:102]:
```suggestion
  - For this specific example. when we implement both patterns it is better to open only one parallel region with `#pragma omp parallel` and use the `pragma omp for` before each loop:
```

**Review by lukasrothenberger — CHANGES_REQUESTED** [review]

**Review by lukasrothenberger — APPROVED** [review]

**Review by lukasrothenberger — APPROVED** [review]

**Pull request merged**